packages/utils/src/formatDate.ts

```diff
@@ -0,0 +1,70 @@
+import { differenceInDays, formatDistanceToNowStrict } from "date-fns"
+import { nb, enUS } from "date-fns/locale"
+
+const DEFAULT_DAYS_RELATIVE_THRESHOLD = 3
+const DEFAULT_LOCALE = "nb-NO"
+
+const dateFNSLocaleFromIntlLocale = {
+  "nb-NO": nb,
+  "en-US": enUS,
+} as const
+
+type SupportedLocales = keyof typeof dateFNSLocaleFromIntlLocale
+
+/**
+ * Formats a date into a string representation. If the date is within the date threshold, the date will be formatted as a relative date.
+ *
+ * @param date - The date to format.
+ * @param options - Optional formatting options.
+ * @param options.forceAbsolute - If set to true, the date will always be formatted using the `formatRemainingTime` function.
+ * @param options.includeTime - If set to true, the time will be included in the formatted date.
+ * @param options.includeWeekday - If set to true, the weekday will be included in the formatted date.
+ * @param options.locale - The locale to use for formatting. Defaults to `nb`.
+ * @param options.relativeDateThresholdDays -
+ *        The number of days to use as a threshold for relative date formatting. Defaults to `3`.
+ *        If the difference between the current date and the provided date is greater than this threshold,
+ *        the date will be formatted using the `formatRemainingTime` function.
+ * @returns The formatted date string.
+ */
+export const formatDate = (
+  date: Date,
+  options?: {
+    forceAbsolute?: boolean
+    includeTime?: boolean
+    includeWeekday?: boolean
+    locale?: SupportedLocales
+    relativeDateThresholdDays?: number
+  }
+): string => {
+  const daysDifference = differenceInDays(new Date(), date)
+  const daysThreshold = options?.relativeDateThresholdDays ?? DEFAULT_DAYS_RELATIVE_THRESHOLD
+  const locale = options?.locale ?? DEFAULT_LOCALE
+
+  if (options?.forceAbsolute !== true && daysThreshold > Math.abs(daysDifference)) {
+    return formatRemainingTime(date, { locale })
+  }
+
+  const formatOptions: Intl.DateTimeFormatOptions = {
+    day: "2-digit",
+    month: "2-digit",
+    year: "numeric",
+    ...(options?.includeWeekday && { weekday: "long" }),
+    ...(options?.includeTime && { hour: "2-digit", minute: "2-digit" }),
+  }
```

Comment:
I'm skeptical of this module bearing its weight vs. just telling people to use Intl.DateTimeFormat / Intl.RelativeDateFormat everywhere—it seems it makes more sense to expose this as a set of default format-options, and avoid having to wrap it ourselves?

Reply:
I understand where you are coming from. At the same time, I think it simplifies the process to just simply put the date into a function that fixes it for you. I can imagine using Intl as a novice could be needless struggle. I could maybe expose the settings as well, for ""advanced"" date formatting

Reply:
I think if we want a standard formatting library we should abstract over Intl